pkg/services/ngalert/state/template_test.go

```diff
@@ -0,0 +1,355 @@
+package state
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana/pkg/services/ngalert/eval"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	ptr "github.com/xorcare/pointer"
+)
+
+func TestTemplateCaptureValueStringer(t *testing.T) {
+	cases := []struct {
+		name     string
+		value    templateCaptureValue
+		expected string
+	}{{
+		name:     "0 is returned as integer value",
+		value:    templateCaptureValue{Value: 0},
+		expected: "0",
+	}, {
+		name:     "1.0 is returned as integer value",
+		value:    templateCaptureValue{Value: 1.0},
+		expected: "1",
+	}, {
+		name:     "1.1 is returned as decimal value",
+		value:    templateCaptureValue{Value: 1.1},
+		expected: "1.1",
+	}}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			assert.Equal(t, c.expected, c.value.String())
+		})
+	}
+}
+
+func TestExpandTemplate(t *testing.T) {
```

Comment:
Are we missing cases for the functions we don't support? I'd like us to test against everything even if their return value is `""`.